src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3732,16 +3732,8 @@ private unsafe void ParseRemaining()
                 return 0;
             }
 
-            // Here could be a possibly valid, and not well-known scheme
-            // Finds the scheme delimiter
-            // we don;t work with the schemes names > c_MaxUriSchemeName (should be ~1k)
-            if ((end - idx) > c_MaxUriSchemeName)
-            {
-                err = ParsingError.SchemeLimit;
-                return 0;
-            }
-
-            //Check the syntax, canonicalize  and avoid a GC call
+            // This appears to be an unknown but potentially valid scheme.
```

Comment:
Is this true? Isn't it this function that's checking for common shemes?

Reply:
Just saw this -- I'll fix that before merging.